web/tests/stream_settings_ui.test.js

```diff
@@ -133,6 +134,9 @@ run_test("redraw_left_panel", ({mock_template}) => {
     function test_filter(params, expected_streams) {
         $("#streams_overlay_container .stream-row:not(.notdisplayed)").length = 0;
         const stream_ids = stream_settings_ui.redraw_left_panel(params);
+        if ($(".stream-row.active").hasClass("notdisplayed")) {
+            stream_edit.empty_right_panel();
+        }
```

Comment:
Why have you added this code? We are not resetting the right panel when filtering.

Reply:
The thing is that earlier `redraw_left_panel` was handling the task to remove `active` class if it had `notdisplayed` in it. So I was overcompensating it by removing the `active` class here. Now that I think about it, maybe we should have `removeClass` for active stream rows in `redraw_left_panel` to handle this.

Reply:
More context: Now it seems `active` class of stream row is not being removed on filter search if search filter doesn't match the active stream row. It is only getting removed in tab switch since we moved the code there.

Reply:
The `active` class is added to the stream row which is being showed in the right side panel. So, now that we are showing the stream in right panel even when it is not present in filtered list, why do we want to remove the `active` class?

Reply:
I would also suggest just to play with the UI a bit to make sure these changes do not introduce another bug like we have two stream names with `active` class, etc.